File: src/libs/jderobotcomm_cpp/src/motorsClient.cpp
#include <jderobot/comm/motorsClient.hpp>
#include <jderobot/comm/ice/motorsIceClient.hpp>
#ifdef JDERROS
#include <jderobot/comm/ros/publisherMotors.hpp>
#endif

namespace JdeRobotComm {

MotorsClient* 
getMotorsClient(Ice::CommunicatorPtr ic, std::string prefix){
	MotorsClient* client = 0;
	Ice::PropertiesPtr prop = ic->getProperties();


	int server = prop->getPropertyAsIntWithDefault(prefix+".Server",0);
	switch (server){
		case 0:
		{
			std::cout << "Motors disabled" << std::endl;
			break;
		}
		case 1:
		{
			std::cout << "Sending Velocities by ICE interfaces" << std::endl;
			MotorsIceClient* cl;
			cl = new MotorsIceClient(ic, prefix);
		    client = (JdeRobotComm::MotorsClient*) cl;
		    break;
		}
		case 2:
		{
			#ifdef JDERROS
				std::cout << "Sending Velocities by ROS messages" << std::endl;
				std::string nodeName;
				nodeName =  prop->getPropertyWithDefault(prefix+".Name","MotorsNode");
				std::string topic;
				topic = prop->getPropertyWithDefault(prefix+".Topic","");
				PublisherMotors* pm;
				pm = new PublisherMotors(0, nullptr, nodeName, topic);
				pm->start();
				client = (JdeRobotComm::MotorsClient*) pm;
            #else
				throw "ERROR: ROS is not available";
			#endif
		 	break;
		}
		default:
		{
			std::cerr << "Wrong " + prefix+".Server property" << std::endl;
			break;
		}

	}

	return client;


}

}//NS

File: src/libs/jderobotcomm_py/jderobotComm/motorsClient.py
import sys
import Ice
import rospy
from .ice.motorsIceClient import MotorsIceClient

if (sys.version_info[0] == 2):
    from .ros.publisherMotors import PublisherMotors

def __getMotorsIceClient(ic, prefix):
    '''
    Returns a Motors Ice Client. This function should never be used. Use getMotorsClient instead of this

    @param ic: Ice Communicator
    @param prefix: prefix name of client in config file

    @type ic: Ice Communicator
    @type prefix: String

    @return Motors Ice Client

    '''
    print("Publishing Motors with ICE interfaces")
    client = MotorsIceClient(ic, prefix)
    client.start()
    return client

def __getPublisherMotors(ic, prefix):
    '''
    Returns a Motors ROS Publisher. This function should never be used. Use getMotorsClient instead of this

    @param ic: Ice Communicator
    @param prefix: prefix name of client in config file

    @type ic: Ice Communicator
    @type prefix: String

    @return Motors ROS Publisher

    '''
    if (sys.version_info[0] == 2):
        print("Publishing Motors with ROS messages")
        prop = prop = ic.getProperties()
        topic = prop.getPropertyWithDefault(prefix+".Topic","");
        client = PublisherMotors(topic)
        return client
    else:
        print(prefix + ": ROS msg are diabled for python "+ sys.version_info[0])
        return None

def __Motorsdisabled(ic, prefix):
    '''
    Prints a warning that the client is disabled. This function should never be used. Use getMotorsClient instead of this

    @param ic: Ice Communicator
    @param prefix: prefix name of client in config file

    @type ic: Ice Communicator
    @type prefix: String

    @return None

    '''
    print(prefix + " Disabled")
    return None

def getMotorsClient (ic, prefix, node=None):
    '''
    Returns a Motors Client.

    @param ic: Ice Communicator
    @param prefix: prefix name of client in config file
    @param node: ROS node

    @type ic: Ice Communicator
    @type prefix: String
    @type node: ROS node

    @return None if Motors is disabled

    '''
    prop = prop = ic.getProperties()
    server = prop.getPropertyAsIntWithDefault(prefix+".Server",0)

    cons = [__Motorsdisabled, __getMotorsIceClient, __getPublisherMotors]

    return cons[server](ic, prefix)